globus_cli/services/transfer/async_transfer.py

```diff
@@ -144,26 +153,26 @@ def async_transfer_command(batch, sync_level, recursive, destination, source,
         label=label, sync_level=sync_level, verify_checksum=verify_checksum,
         preserve_timestamp=preserve_mtime, encrypt_data=encrypt)
 
-    if source_path and dest_path:
-        transfer_data.add_item(source_path, dest_path,
-                               recursive=recursive)
     if batch:
         @click.command()
         @click.option('--recursive', is_flag=True)
-        @click.argument('source_path')
-        @click.argument('dest_path')
+        @click.argument('source_path', type=TaskPath(base_dir=cmd_source_path))
+        @click.argument('dest_path', type=TaskPath(base_dir=cmd_dest_path))
         def process_batch_line(dest_path, source_path, recursive):
             """
             Parse a line of batch input and turn it into a transfer submission
             item.
             """
-            transfer_data.add_item(source_path, dest_path,
+            transfer_data.add_item(str(source_path), str(dest_path),
                                    recursive=recursive)
 
         shlex_process_stdin(
             process_batch_line,
             ('Enter transfers, line by line, as\n\n'
-             '    [--recursive] source-path dest-path\n'))
+             '    [--recursive] SOURCE_PATH [DEST_PATH]\n'))
```

Comment:
Need to drop the brackets indicating optionalness here. (Maybe fixup 24de9fdfcd7b52e626da49af0ef9ce4bdc50e410?)